### ISSUE TITLE
Adjusting trigger sensitivity for Xinput pads

### ISSUE THREAD

**Issue by Krusher97** [m64p/mupen64plus-GLideN64#24]
Right now the triggers on my Xinput pad only seem to register when at 100% pressure. Is there any way to make it so that they register at 80%?

**Comment by loganmc10** [comment]
Do the triggers register as an axis or a button in the controller dialog?

**Comment by Krusher97** [comment]
As an axis

**Comment by loganmc10** [comment]
Are you certain that your triggers are pressure sensitive? I have an 3rd party Xinput controller, and the trigger registers as an axis, but its not pressure sensitive. It's just either on or off (the axis either registers as -32768 or 32767, but nothing in between).

**Comment by Krusher97** [comment]
![axis](https://user-images.githubusercontent.com/25664221/31781736-d282d5fc-b516-11e7-9986-b517d4940db6.png)
It seems to be.

**Comment by loganmc10** [comment]
This is a little tricky for me to test, it could be that SDL doesn't support pressure sensitive triggers.

Tonight I'll try out my Xbox controller, I assume it's triggers are pressure sensitive.

Basically the axis's max value is 32768, and the deadzone for axis->button presses like you are attempting is set to 16384 in the input plugin, so it should trigger a press at 50% pressure, unless SDL isn't properly reporting the pressure.

**Comment by loganmc10** [comment]
I just tested this using a first party Xbox One controller in Linux. It does respond to the pressure. The axis actually reports from -32768 to 32767, so since the deadzone is at 16384, it does respond at around 75% pressure.

I tested this in Super Mario 64 and it worked as expected, he crouches at around 75% pressure.

So its hard for me to say what might be up with your controller, my guess would be that maybe SDL isn't able to detect the pressure of the trigger. I'm not sure what I would be able to do, the code is pretty simple, it's basically:

```Get axis reading from SDL. If axis reading > deadzone (16384) then register a button press```

**Comment by Krusher97** [comment]
Maybe an ability to modify the pressure sensitivity so that the button is triggered at whatever value you put in instead of fixed at 75%?
I guess that would be harder to implement without breaking other things.

**Comment by loganmc10** [comment]
The input plugin has a way to configure the deadzone per axis in these situations, not currently exposed by the GUI.

The problem is SDL. For your right analog stick, the axis value goes from 0 to 32768, so 16384 represents 50%. For the trigger, the axis goes from -32768 to 32768, so 0 represents 50%.

There is no way for me to determine the difference between the trigger and an analog stick, they are both just an axis, but they have different minimum values. So I don't know of a way to allow the user to modify the deadzone, and be able to convey to them what the deadzone value really represents

**Comment by Krusher97** [comment]
Do you think this problem could be related to https://github.com/mupen64plus/mupen64plus-user-issues/issues/695?

The triggers are treated as an axis so it could be that they aren't getting their full range of motions either.

**Comment by loganmc10** [comment]
No, they are unrelated, that range limiting function is only applied when an axis is mapped to the N64 analog stick, not if its mapped to a button or trigger

**Comment by Papermanzero** [comment]
Antimicro is also sdl based. They managed to adjust the sensitivity of the triggers. Maybe the project could be helpful for this request.
